app/models/event.rb

```diff
@@ -41,6 +41,7 @@ class Event < ActiveRecord::Base
   scope :canceled, -> { where(state: 'canceled') }
   scope :withdrawn, -> { where(state: 'withdrawn') }
   scope :highlighted, -> { where(is_highlight: true) }
+  scope :current, -> { joins(:event_type).where("start_time <= ? AND datetime(strftime('%s', start_time) + event_types.length * 60,  'unixepoch') >= ?", Time.current, Time.current) }
```

Comment:
Uff, but already expected sth like this :(
I'll fiddle a little bit around with it, maybe we can use some Rails magic to accomplish the same...